controller/space.go

```diff
@@ -53,6 +54,26 @@ func (c *SpaceController) Create(ctx *app.CreateSpaceContext) error {
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
+		/*
+			Should we create the new area
+			- over the wire(service) something like app.NewCreateSpaceAreasContext(..), OR
+			- as part of a db transaction ?
+
+			The argument 'for' creating it at a transaction level is :
+			You absolutely need both space creation + area creation
+			to happen in a single transaction as per requirements.
+		*/
+
+		newArea := area.Area{
+			ID:      satoriuuid.NewV4(),
+			SpaceID: space.ID,
+			Name:    space.Name,
+		}
+		err = appl.Areas().Create(ctx, &newArea)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
```

Comment:
It might be more descriptive if we return the error like this:
```
			return jsonapi.JSONErrorResponse(ctx, errs.Wrap(err, fmt.Sprintf("Error creating area")))
```